poltergeist/result.py

```diff
@@ -31,13 +31,13 @@ def unwrap_or(self, default: DefaultT) -> T:
     def unwrap_or(self, default: Any = None) -> Any:
         return self.unwrap()
 
-    def unwrap_or_else(self, op: Callable[[E], DefaultT]) -> T:
+    def unwrap_or_else(self, op: Any) -> T:
```

Comment:
Using Any as the argument's type works, but breaks type inference when using Result.unwrap_or_else() with lambdas. I tried to use NoReturn instead, but MyPy was complaining. So I think this is the best approach for now.